src/test/java/io/polivakha/mojo/properties/ReadPropertiesMojoTest.java

```diff
@@ -1,37 +1,126 @@
+        Assertions.assertThat(secondValue).isEqualTo("9098");
+        Assertions.assertThat(nonExistingProperty).isNull();
+    }
+
+    @Test
+    public void givenSeveralPropertyFiles_whenReadingFromBoth_thenPropertiesFromAllFilesRead() throws IOException, MojoExecutionException, MojoFailureException {
+        Path tempDirectory = Files.createTempDirectory("mytmp");
+        Path firstFile = Files.createFile(Paths.get(tempDirectory.toString(), "first.properties"));
+        Path secondFile = Files.createFile(Paths.get(tempDirectory.toString(), "second.properties"));
+
+        try (FileWriter writer = new FileWriter(firstFile.toFile())) {
+            writer.write("first.version=1.0.1\n");
+            writer.write("second.version=1.4.2");
+        }
+
+        try (FileWriter writer = new FileWriter(secondFile.toFile())) {
+            writer.write("third.version=5.8.8\n");
+        }
+
+        readPropertiesMojo.setIncludes(new String[]{Path.of(tempDirectory.toString(), "*.properties").toString()});
+        readPropertiesMojo.execute();
+
+        String one = projectStub.getProperties().getProperty("first.version");
+        String two = projectStub.getProperties().getProperty("second.version");
+        String three = projectStub.getProperties().getProperty("third.version");
+
+        Assertions.assertThat(one).isEqualTo("1.0.1");
+        Assertions.assertThat(two).isEqualTo("1.4.2");
+        Assertions.assertThat(three).isEqualTo("5.8.8");
+    }
+
+    @Test
+    public void givenTwoPropertiesFileWithTwoPatterns_whenResolvingExistingProperties_thenLastPatternFilesTakePrecedence() throws IOException, MojoExecutionException, MojoFailureException {
+        Path parentFirst = Files.createTempDirectory("parent1");
+        Path parenSecond = Files.createTempDirectory("parent2");
+
+        Path firstFile = Files.createFile(Paths.get(parentFirst.toString(), "first.properties"));
+        Path secondFile = Files.createFile(Paths.get(parenSecond.toString(), "second.properties"));
+
+        try (FileWriter writer = new FileWriter(firstFile.toFile())) {
+            writer.write("first.version=1.0.1\n");
+            writer.write("second.version=1.4.2");
+        }
+
+        try (FileWriter writer = new FileWriter(secondFile.toFile())) {
+            writer.write("third.version=5.8.8\n");
+            writer.write("first.version=0.0.1\n");
+        }
+
+        readPropertiesMojo.setIncludes(
+          new String[]{
+            Path.of(parentFirst.toString(), "*.properties").toString(),
+            Path.of(parenSecond.toString(), "*.properties").toString()
+          }
+        );
+
+        readPropertiesMojo.execute();
+
+        String one = projectStub.getProperties().getProperty("first.version");
+        String two = projectStub.getProperties().getProperty("second.version");
+        String three = projectStub.getProperties().getProperty("third.version");
+
+        Assertions.assertThat(one).isEqualTo("0.0.1");
```

Comment:
if i understand right it just work method put in HashTable when we do properties.put() method and this logic from there and pattern logic test is not relevant for it. Or do you want to be very strongly confident in the work of the method?)

Reply:
I do not understand anything, can you please elaborate a bit more on your question?)

Reply:
I don't say that this test is bad, no) it's just result of observation that this logic is obviously because of properties.put() method do it inside itself. But if we just want to check that our logic doesn't break previous logic then ok, but i think that it will not break)

Reply:
I do not know to be honest how it was done in original version) That behavior seemed to be natural and pretty logical